src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/CompositeSearchValue.cs

```diff
@@ -65,5 +65,11 @@ public override string ToString()
         {
             return string.Join(" $ ", Components.Select(component => string.Join(", ", component.Select(v => $"({v})"))));
         }
+
+        /// <inheritdoc />
+        public override int GetHashCode()
+        {
+            return Components.GetHashCode();
```

Comment:
>Components [](http://example.com/codeflow?start=19&length=10)

I'm afraid it will count hashcode not by combination of collection entities, but by memory address of collection.

Reply:
should be HashCode.Combine(Components.Select(x=>x.GetHashCode())